plugin/documents.py

```diff
@@ -213,13 +213,16 @@ def session_buffers_async(self) -> Generator[SessionBuffer, None, None]:
             yield sv.session_buffer
 
     def on_text_changed_async(self, changes: Iterable[sublime.TextChange]) -> None:
-        self._clear_highlight_regions()
         different, current_region = self._update_stored_region_async()
         if self.view.is_primary():
             for sv in self.session_views_async():
                 sv.on_text_changed_async(changes)
         if not different:
             return
+        if "documentHighlight" not in global_settings.disabled_capabilities:
+            self._clear_highlight_regions()
+            self._when_selection_remains_stable_async(self._do_highlights, current_region,
+                                                      after_ms=self.highlights_debounce_time)
```

Comment:
Duplicates existing code. Makes sense to create a helper function for it.

Reply:
I would leave it as it is currently. 
But I will add a new function that will wrap this code if you really think it is needed. (in that case could you give me a name for that function, because I can't think of any good one, `schedule_highlight` was the best one I could think of) :)

Reply:
Since one place checks `is_in_higlighted_region` and one not, I'm no longer sure if it's worth creating a helper for them.